main.js

```diff
@@ -196,6 +196,77 @@ document.addEventListener('keydown', (event) => {
   }
 });
 
+/* Contact Validation */
+const form = document.querySelector('.form');
+const fullNameInput = form.full_name;
+const firstNameInput = form.first_name;
+const lastNameInput = form.last_name;
+const messageInput = form.message;
+const errorElement = document.querySelector('.validation-message');
+const errorMessage = 'The email address should be in lowercase.';
+const emailInput = form.email;
+const submitBtn = form.submit;
+
+submitBtn.addEventListener('click', (event) => {
+  if (emailInput.value !== emailInput.value.toLowerCase()) {
+    errorElement.textContent = errorMessage;
+    event.preventDefault();
+  }
+  if (window.innerWidth > 376) {
+    firstNameInput.removeAttribute('required');
+    lastNameInput.removeAttribute('required');
+  } else {
+    fullNameInput.removeAttribute('required');
+  }
+});
+
+/* Functionaliy to preserve data in local storage */
+const formData = {
+  fullName: '',
+  firstName: '',
+  lastName: '',
+  email: '',
+  message: '',
+};
+
+if (localStorage.formDataStorage) {
+  formData.fullName = JSON.parse(localStorage.formDataStorage).fullName;
+  formData.firstName = JSON.parse(localStorage.formDataStorage).firstName;
+  formData.lastName = JSON.parse(localStorage.formDataStorage).lastName;
+  formData.email = JSON.parse(localStorage.formDataStorage).email;
+  formData.message = JSON.parse(localStorage.formDataStorage).message;
+  fullNameInput.value = formData.fullName;
+  firstNameInput.value = formData.firstName;
+  lastNameInput.value = formData.lastName;
+  emailInput.value = formData.email;
+  messageInput.value = formData.message;
+}
+
+fullNameInput.addEventListener('input', () => {
+  formData.fullName = fullNameInput.value;
+  localStorage.setItem('formDataStorage', JSON.stringify(formData));
+});
+
+firstNameInput.addEventListener('input', () => {
+  formData.firstName = firstNameInput.value;
+  localStorage.setItem('formDataStorage', JSON.stringify(formData));
+});
+
+lastNameInput.addEventListener('input', () => {
+  formData.lastName = lastNameInput.value;
+  localStorage.setItem('formDataStorage', JSON.stringify(formData));
+});
+
+emailInput.addEventListener('input', () => {
+  formData.email = emailInput.value;
+  localStorage.setItem('formDataStorage', JSON.stringify(formData));
+});
+
+messageInput.addEventListener('input', () => {
+  formData.message = messageInput.value;
+  localStorage.setItem('formDataStorage', JSON.stringify(formData));
```

Comment:
- [ ] Make sure When the user loads the page, the code checks if there are any saved data in the local storage, and if so the input fields are pre-filled with the data